### ISSUE TITLE
dasel complete jq replacement? Problem with maps

### ISSUE THREAD

**Issue by chairraver** [TomWright/dasel#52]
**Describe the desired outcome**
Nice tool, but I'm having a problem to achieve a particular output. Assuming the following input:
```
{   
  "Admin": {                                                                                                                    
    "ClientID": "eBZ5kp",         
    "ClientSecret": "rSiGsF"
  },                     
  "User": {       
    "ClientID": "Y1Btrp",
    "ClientSecret": "pHfeK-"
  }
}
```
I want to use the ClientIDs in a subsequent dasel search on another file with ClientID as the search item (with some shell glue). I wasn't able  to achieve the desired results with dasel, which I could achieve fairly easy with jq:
```
jq -r '.[].ClientID'
eBZ5kp
Y1Btrp
```

**Attempts**
dasel -s '.' reproduced the input file, but I could not achieve the same result as with jq. 

I went through the docs a couple of times, but I didn't saw anything that would result in the desired outcome. I typically would see errors like:
```
dasel select -f /tmp/clients.json -p json -s '.ClientID
Error: could not query node: could not find value: no value found for selector: .ClientID: map[......
```
and various combinations of ., [], [*] and .ClientID. Maybe I missed something obvious.

**Comment by TomWright** [comment]
You can use the search selector, looking for a key value of `ClientID`:
```
echo '{   
  "Admin": {
    "ClientID": "eBZ5kp",
    "ClientSecret": "rSiGsF"
  },
  "User": {
    "ClientID": "Y1Btrp",
    "ClientSecret": "pHfeK-"
  }
}' | dasel -p json -m --plain '.(?:keyValue=ClientID)'
eBZ5kp
Y1Btrp
```

The `[]` and `[*]` selectors are only usable with arrays. If this were an array of objects then you could do something like `.[*].ClientID`.

I could look at making the `[*]` selector also support map values in the future. If you need that then let me know 🙂

**Comment by TomWright** [comment]
It's actually quite easy to add the any index selector and dynamic selector for maps as well as arrays so I'll go ahead and implement that

**Comment by chairraver** [comment]
Hey, thanks a lot.

First of all I'm on Arch Linux and when I installed I only got version 1.6.2. I have now updated to 1.7.0.

And yes indeed I didn't use the keyValue search thing. And it could have been the absence of -m also played a role.

I'm happy now, thanks again.

**Comment by TomWright** [comment]
The search selector docs are here: https://github.com/TomWright/dasel#search

There is a basic example with key based searching there already but let me know if you don't think that's good enough and I can update the readme.

**Comment by chairraver** [comment]
No, I think everything is good. I remember trying (?:-, but I guess the implications of keyValue/-  didn't really sink in (and possibly the absence of the -m flag).  So alas, during trying I simply didn't hit the right combination.

**Comment by TomWright** [comment]
OK good to hear. On a side note I've just released `v1.8.0` which will allow you to use a simpler selector of `.[*].ClientID`.
